nikola/post.py

```diff
@@ -843,7 +859,12 @@ def permalink(self, lang=None, absolute=False, extension='.html', query=None):
             extension = self.compiler.extension()
 
         pieces = self.translations[lang].split(os.sep)
-        pieces += self.folder.split(os.sep)
+        folder = self.meta[lang].get('path', self.folder_relative)
```

Comment:
Make this an instance variable instead of copy-pasting this around and potentially running it multiple times.